pkg/operator/encryption/deployer/unionrevisionedpod.go

```diff
@@ -36,6 +36,9 @@ func (d *UnionRevisionLabelPodDeployer) DeployedEncryptionConfigSecret() (secret
 		if !converged || err != nil {
 			return nil, converged, err
 		}
+		if secret == nil {
```

Comment:
I will use it in `OAS-O` - the custom `deployer` will not return any resources when the annotation was `set`.

Reply:
What do we see if the oauth-apiserver secret does not exist?

Reply:
the same - https://github.com/openshift/library-go/blob/master/pkg/operator/encryption/deployer/revisionedpod.go#L116

Reply:
```
// if encryption is not enabled at this revision or the secret was deleted, we should not error
if errors.IsNotFound(err) {
	return nil, true, nil
}
```

Reply:
My thought was that we might want to block COAO from progressing if its secret exists, but CAO's secret does not.

Reply:
if https://github.com/openshift/cluster-openshift-apiserver-operator/pull/348 makes sense then we need to merge this pull.